modules/database/src/ioc/dbStatic/dbLexRoutines.c

```diff
@@ -1045,9 +1061,14 @@ static void dbRecordHead(char *recordType, char *name, int visible)
         yyerrorAbort("dbRecordHead: Record name can't be empty");
         return;
     }
-    badch = strpbrk(name, " \"'.$");
+    badch = strpbrk(name, " \t\"'.$");
     if (badch) {
-        epicsPrintf("Bad character '%c' in record name \"%s\"\n",
+        epicsPrintf("Error: Bad character '%c' in record name \"%s\"\n",
+            *badch, name);
+        yyerrorAbort(NULL);
+        return;
+    } else if((*name >= '0' && *name <= '9') || *name=='{') {
+        epicsPrintf("Warning: Bad character '%c' begins record name \"%s\"\n",
             *badch, name);
```

Comment:
Shouldn't this now be *name, not *badch?

Reply:
Yes, it should.  Good catch.